artifactory/commands/transferinstall/installplugin.go

```diff
@@ -0,0 +1,356 @@
+package transferinstall
+
+import (
+	"fmt"
+	downloadutils "github.com/jfrog/build-info-go/utils"
+	"github.com/jfrog/gofrog/version"
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	"github.com/jfrog/jfrog-client-go/artifactory"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/pkg/errors"
+	"net/http"
+	url "net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	pluginReloadRestApi = "api/plugins/reload"
+	jfrogHomeEnvVar     = "JFROG_HOME"
+	latest              = "[RELEASE]"
+)
+
+var (
+	defaultSearchPath = filepath.Join("opt", "jfrog")
+	// Plugin directory locations
+	originalDirPath = FileItem{"artifactory", "etc", "plugins"}
+	v7DirPath       = FileItem{"artifactory", "var", "etc", "artifactory", "plugins"}
+	// Error types
+	emptyUrlErr            = errors.Errorf("Base download URL must be provided to allow file downloads.")
+	notValidDestinationErr = errorutils.CheckErrorf("Can't find the directory in which to install the data-transfer plugin. Please ensure you're running this command on the machine on which Artifactory is installed. You can also use the --home-dir option to specify the directory.")
+	downloadConnectionErr  = func(baseUrl string) error {
+		return errorutils.CheckErrorf("We tried to download the plugin files from '%s' but got connection issue. Hint: manual transfer the files to the machine and use --source-dir option.", baseUrl)
+	}
+)
+
+type InstallPluginCommand struct {
+	// The name of the plugin the command will install
+	pluginName string
```

Comment:
I implemented this to provide a basic service for installing Artifactory plugin i.e. downloading/copy files into the Artifactory 'plugins' directory and request a reload after the transfer.
all you need to specify is the name, downloadURL and the files of the plugin as an argument and the service will take care of the rest.
it will make it easier to create install command for other plugins as well.